### PR TITLE
Support string FQN as a way to add lineage information

### PR DESCRIPTION
**Pull request by Abacn**
Python SDK counterpart of #32585

**Please** add a meaningful description for your change here

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

File: sdks/python/apache_beam/metrics/metric.py [hunk 1]
```diff
@@ -370,7 +370,27 @@ def get_fq_name(
...
+    - `add("system", "segment1", "segment2")`
+    - `add("system", "segment1", "segment2", subtype="subtype")`
+
+    Example of adding a FQN:
+
+    - `add("system:segment1.segment2")`
+    - `add("system:subtype:segment1.segment2")`
+    """
+    if len(segments) == 0 and subtype is None:
+      self.metric.add(system)
```
Comment (Abacn) [inline sdks/python/apache_beam/metrics/metric.py:391]: strictly speaking this is "system_or_fqn" however naming change may break positional parameters so the original name was kept.

**Comment by Abacn** [comment]
R: @rohitsinha54

File: sdks/python/apache_beam/metrics/metric.py (hunk 1 above)
Comment (rohitsinha54) [reply sdks/python/apache_beam/metrics/metric.py:391]: It appears we are overloading the system parameter which can be confusing can we add a separate optional parameter for fqn string say "details"

details=None

    if details:
      self.metric.append(details)  
    elif system and segments:
      self.metric.append(self.get_fq_name(system, subtype, segments))  # Add the constructed FQN

Comment (Abacn) [reply sdks/python/apache_beam/metrics/metric.py:391]: then one has to call `add(details=...)` to access to this branch which is less preferable either. Let me assign it into an intermediate variable (`system_or_fqn = system`)

**Review by rohitsinha54 — APPROVED** [review]

**Pull request merged**